test/unit_test/character_sheet_unit_tests/TalentTests.java

```diff
@@ -0,0 +1,170 @@
+package unit_test.character_sheet_unit_tests;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import rpg_database.character_sheet.BaseClasses;
+import rpg_database.character_sheet.Fields;
+import rpg_database.character_sheet.Focuses;
+import rpg_database.character_sheet.Talent;
+import rpg_database.character_sheet.TalentLevels;
+import rpg_database.character_sheet.Talents;
+import rpg_database.character_sheet.WeaponGroups;
+
+public class TalentTests {
+
+	// test setup
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	// test methods
+	@Test
+	public void testGetDefaultTalentLevel() {
+		Talent talent = new Talent(Talents.ALCHEMY);
+		assertEquals(TalentLevels.NOVICE, talent.getTalentLevel());
+	}
+
+	@Test
+	public void testGetMasterTalentLevel() {
+		Talent talent = new Talent(Talents.ALCHEMY, TalentLevels.MASTER);
+		assertEquals(TalentLevels.MASTER, talent.getTalentLevel());
+	}
+
+	@Test
+	public void testSetMasterTalentLevel() {
+		Talent talent = new Talent(Talents.ALCHEMY, TalentLevels.MASTER);
+		talent.setTalentLevel(TalentLevels.JOURNEYMAN);
```

Comment:
Az jó nekünk, hogy engedi a rendszer visszafelé downgradelni a talentet? Egy issue erre is kellene, mert ha igen akkor itt is resetelni kellene például a felvett specializációt.

Reply:
A felvett specializáció novice-ra downgradelve is fel van véve. Nincs mit resetelni. Nem értem a problémát.

Viszont jó hogy commenteltél ide mert a teszt nevét nem fejeztem be... Azt át kell írnom setMasterTalentLevelToLowerLevel-re. Ilyenekre is nyugodtan kommentelhetsz

Reply:
Későbbi task *telefon